packages/itwinui-react/src/core/Menu/MenuItem.tsx

```diff
@@ -106,28 +116,69 @@ export const MenuItem = React.forwardRef((props, forwardedRef) => {
     ...rest
   } = props;
 
+  const parent = React.useContext(MenuItemContext);
+
   const menuItemRef = React.useRef<HTMLElement>(null);
-  const [focusOnSubmenu, setFocusOnSubmenu] = React.useState(false);
   const submenuId = useId();
 
   const [isSubmenuVisible, setIsSubmenuVisible] = React.useState(false);
-  const [isNestedSubmenuVisible, setIsNestedSubmenuVisible] =
+  const [activeIndex, setActiveIndex] = React.useState<number | null>(null);
+
+  const [hasFocusedNodeInSubmenu, setHasFocusedNodeInSubmenu] =
     React.useState(false);
-  const parent = React.useContext(MenuItemContext);
 
-  const onVisibleChange = (open: boolean) => {
-    setIsSubmenuVisible(open);
+  const nodeId = useFloatingNodeId();
+  const tree = useFloatingTree();
+  const parentId = useFloatingParentNodeId();
+  const parentTreeIndex =
+    menuItemRef.current?.dataset['menuChildIndex'] != null
+      ? Number(menuItemRef.current?.dataset['menuChildIndex'])
+      : undefined;
 
-    // we don't want parent to close when mouse goes into a nested submenu,
-    // so we need to let the parent know whether the submenu is still open.
-    parent.setIsNestedSubmenuVisible(open);
-  };
+  React.useEffect(() => {
+    const handleNodeFocused = (event: TreeEvent) => {
+      if (
+        // Consider a node "X" with its submenu "Y".
+        // Focusing "X" should close all submenus of "Y".
+        parentId === event.nodeId ||
+        // When a node "X" is focused, close "X"'s siblings' submenus
+        // i.e. only one submenu in each menu can be open at a time
+        (parentId === event.parentId && nodeId !== event.nodeId)
+      ) {
+        setIsSubmenuVisible(false);
+        setHasFocusedNodeInSubmenu(false);
+      }
+    };
+
+    tree?.events.on('nodeFocused', handleNodeFocused);
+
+    return () => {
+      tree?.events.off('nodeFocused', handleNodeFocused);
+    };
+  }, [nodeId, parentId, tree?.events, tree?.nodesRef, children]);
+
+  const listItemsRef = React.useRef<Array<HTMLElement | null>>([]);
 
   const popover = usePopover({
-    visible: isSubmenuVisible || isNestedSubmenuVisible,
-    onVisibleChange,
+    nodeId,
+    visible: isSubmenuVisible,
+    onVisibleChange: setIsSubmenuVisible,
     placement: 'right-start',
-    trigger: { hover: true, focus: true },
+    interactions: {
+      hover: !hasFocusedNodeInSubmenu, // If focus is still inside submenu, don't close the submenu upon hovering out.
+      listNavigation: true,
+    },
+    interactionsProps: {
+      hover: {
+        move: false,
```

Comment:
Would this be better suited as the default for `usePopover`? Do we ever need `move: true`?

Reply:
I thought to avoid that to prevent causing any behavior changes for users already used to `move: true`. Also, I'm not sure in which cases would `move: true` be needed. But since Floating UI has set that as the default, I guess there is some reason for it?

I can add an after PR TODO to address this comment later or I can make the change in this PR itself if we think it won't impact users.

Reply:
Since it's a very small change, I'm ok with doing it in the same PR. Just make sure to check that there is no adverse behavioral effect, and add a short changeset as well.

Reply:
Sounds good, I made that change in this PR (https://github.com/iTwin/iTwinUI/pull/1942/commits/93e5a33eed228c1cb43da0b50da9e4db062dc30d).

Since we don't expose the hover trigger in the public `Popover` and since we use the private `usePopover`'s `hover` trigger only in `MenuItem`, I think the existing changeset is sufficient ([code](https://github.com/iTwin/iTwinUI/pull/1942/files#diff-89ddc481dba81a1393f7122d869df11bb687a93faec74c18833d53f9399b297aR5)).